packages/react-router/src/link.tsx

```diff
@@ -119,7 +119,8 @@ export type SearchPaths<
   TSearchPath extends string,
   TPaths = ResolvePaths<TRouter, TSearchPath>,
   TPrefix extends string = `${RemoveTrailingSlashes<TSearchPath>}/`,
-> = TPaths extends `${TPrefix}${infer TRest}` ? TRest : never
+  TFilteredPaths = TPaths & `${TPrefix}${string}`,
```

Comment:
This basically filters down the union with an intersection before distributing over it. Turns out its much faster